runtime-modules/governance/src/council.rs

```diff
@@ -75,6 +85,73 @@ impl<T: Trait> Module<T> {
         CouncilMint::<T>::put(mint_id);
         Ok(mint_id)
     }
+
+    fn add_reward_relationship(
+        destination: &T::AccountId,
+    ) -> Result<T::RewardRelationshipId, recurringrewards::RewardsError> {
+        if let Some(reward_source) = Self::council_mint() {
+            let recipient = <recurringrewards::Module<T>>::add_recipient();
+
+            let next_payout_at = system::Module::<T>::block_number()
+                + Self::first_payout_after_term_start()
+                + T::BlockNumber::one();
+
+            let relationship_id = <recurringrewards::Module<T>>::add_reward_relationship(
+                reward_source,
+                recipient,
+                destination.clone(),
+                Self::amount_per_payout(),
+                next_payout_at,
+                Self::payout_interval(),
+            )?;
+
+            RewardRelationships::<T>::insert(destination, relationship_id);
+
+            Ok(relationship_id)
+        } else {
+            Err(recurringrewards::RewardsError::RewardSourceNotFound)
+        }
+    }
+
+    fn remove_reward_relationships() {
+        for seat in Self::active_council().into_iter() {
+            if RewardRelationships::<T>::exists(&seat.member) {
+                let id = Self::reward_relationships(&seat.member);
+                <recurringrewards::Module<T>>::remove_reward_relationship(id);
+            }
+        }
+    }
+
+    fn apply_council_seats(seats: Seats<T::AccountId, BalanceOf<T>>, term: T::BlockNumber) {
+        <ActiveCouncil<T>>::put(seats.clone());
+
+        let next_term_ends_at = <system::Module<T>>::block_number() + term;
+
+        <TermEndsAt<T>>::put(next_term_ends_at);
+
+        for seat in seats.iter() {
+            let reward_added_result = Self::add_reward_relationship(&seat.member);
+
+            if reward_added_result.is_err() {
```

Comment:
What kind of error is this? what can cause?

Reply:
`recurringrewards::RewardsError`

Since the council mint is an `Option<MintId>` it could fail with `recurringrewards::RewardsError::RewardSourceNotFound`

Reply:
- This now appears to have been rewritten so as to just log if there is no mint set, or if the call to `<recurringrewards::Module<T>>::add_reward_relationship` fails. Pre-sumably this is because either error means the runtime is broken. That choice is fine with me.

- I don't think having an optional mint makes sense, because it does not reflect what is actually going on. But even beyond this, having a non-staic relationship with the mint itself -  let alone having a mutable mint id in stroage - does not make sense, because we do indeed statically want the council to have a mint. I think this is a deeper issue that we need to explore at a later time, perhaps the way we are using mints & recurring rewards is actually not that great. Will open some issue(s) on this later.